cmd/minikube/cmd/start.go

```diff
@@ -1072,6 +1072,10 @@ func validateFlags(cmd *cobra.Command, drvName string) {
 	validateCPUCount(drvName)
 
 	if cmd.Flags().Changed(memory) {
+		if !oci.HasMemoryCgroup() {
```

Comment:
this should be done only if it is Docker or Podman
if driver.IsKIC(drvName) && !oci.HasMemoryCgroup() {

...

}

Reply:
If we add this, then we get 
```
$ make lint
cmd/minikube/cmd/start.go:1054:1: cyclomatic complexity 31 of func `validateFlags` is high (> 30) (gocyclo)
```
So create a helper function to solve this.